### DISCUSSION TITLE
Negative sampling doesn't seem to be working for contextual models

### DISCUSSION THREAD

**Discussion by Malitem** [RUCAIBox/RecBole#1275] [Q&A]
I have this kind of interactions table:
```
user_id:token | item_id:token | label:token
--------------| --------------- | -----------
103615        | 19667           | 1
103615        | 5341            | 1
103615        | 20246           | 1
```
Here all the values in the label column are equal to 1. My understanding of the negative sampling parameter is that it would take this label 1s and would sample one negative item for each positive item. I am running a FM model with the following settings:

> #### dataset config
> USER_ID_FIELD: user_id
> ITEM_ID_FIELD: item_id
> load_col:
>     inter: [user_id, item_id, label]
>     item: [feat1, feat2, item_id]
>     user: [user_id, feat3, feat4, feat5]
> 
> selected_features: [feat1, feat2, feat3, feat4, feat5]
> 
> LABEL_FIELD: label
> 
> neg_sampling: 
>     uniform: 1
> 
> #### model config
> eval_step: 1
> 
> #### Training an evaluation config
> epochs: 10
> train_batch_size: 4096
> eval_batch_size: 900718000
> eval_args:
>     group_by: user
>     order: RO
>     split: {'RS':[0.8,0.2,0]}
>     mode: labeled
> metrics: ['AUC', 'LogLoss']
> valid_metric: AUC

This gives me the following warning: WARNING No negative samples in y_true, false positive value should be meaningless recbole and the evaluation equal to nan.

You can probably guess that this label column was added by me manually. I have tried running the negative sampling without it but it didn't work. Specifically, since I have a **mode: 'labeled'** the error would pop up that there is no label feature. If I switch the mode to full, I get the error saying that the metrics are not made for the mode full. Yet I need these metrics since there is no ranking in my dataset.

There is probably something that I am doing wrong although I cannot find what. Some help would be much appreciated.

**Comment by leoleojie** [comment]
@Malitem sorry for the late reply.
You need to modify the recommendation scenario. 
In particular, context-aware does not support negative sampling, because we can't construct the context information of negative samples(e.g. click time).  Therefore,  instead of setting label column to 1, you need to provide the negative samples.
